query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -140,21 +140,27 @@ extends Iterator[R] with StrictLogging {
     // Lazy variables have an extra lookup cost, due to a volatile bitmap field generated by
     // the compiler. Copy to a local variable to reduce some overhead.
     val wit = windowIt
+    val timestampColID = RangeVector.timestampColID
 
     wit.nextWindow()
     while (wit.hasNext) {
-      val queryInfo = wit.next
-      val nextInfo = ChunkSetInfo(queryInfo.infoPtr)
+      val nextInfo = wit.next
+      val tsVectorBase = nextInfo.vectorBase(timestampColID)
+      val tsVectorOffset = nextInfo.vectorOffset(timestampColID)
+      val tsReader = rv.partition.chunkReader(timestampColID, tsVectorBase, tsVectorOffset).asLongReader
+      val valueVectorBase = nextInfo.vectorBase(rv.valueColID)
+      val valueVectorOffset = nextInfo.vectorOffset(rv.valueColID)
+      val valueReader = rv.partition.chunkReader(rv.valueColID, valueVectorBase, valueVectorOffset)
```

Comment:
Forgot to add one last but very important comment:
Before, the queryInfo instance cached the reader across all windows, for the same chunk.  What you're doing here, is to instead create a new reader with every window.   This is not OK for several reasons:

1) Some readers, especially the Histogram one, is actually a new instance as it contains state.   We don't want to allocate a new HistogramReader instance for every window, only once per chunk.
2) The discovery of the reader is not free, it does involve some cycles.  Caching the reader instance is a query optimization that we don't want to lose.

One idea to move back to being able to cache the reader is to move the reader into the `ChunkSetInfoT`.  This allows the old caching to continue to work.

Reply:
It is not elegant, but I have introduced vars in the ChunkSetInfoT trait that can be used to introduce state into same object without extra allocation.

In fact, current solution now eliminates the per info Long allocation introduced by ChunkInfoIterators.